docs/getting-started/smart-component.md

```diff
@@ -2,21 +2,92 @@
 kind: '🏡 Getting Started'
 title: 'Smart components'
 ---
-
 # What is a smart component?
 
-Smart components are classic Web Components, but they have something special: they communicate with the outside.
+Smart components are Web Components that wrap UI components to create a clear separation between API and UI logic.
+
+While UI components focus solely on rendering and user interactions, smart components handle all the complexity of data fetching, state management, and API interactions automatically.
+
+This architecture ensures a clean separation of concerns where smart components manage the data layer while UI components remain focused on presentation.
+
+Smart components automatically:
+- Fetch initial data when mounted,
+- Handle loading and error states,
+- Update the UI when data changes,
+- Make API calls in response to user actions,
+- Emit events for important state changes,
+- Fire events that a `cc-toaster` component can react to by showing notifications about API call results.
```

Comment:
suggestion: maybe we could leave a link to the component

Reply:
Good idea! :+1: